src/eckit/io/PooledFile.cc

```diff
@@ -26,7 +26,8 @@ namespace eckit {
 
 class PoolFileEntry;
 
-static thread_local std::map<PathName, std::unique_ptr<PoolFileEntry>> pool_;
+static std::map<PathName, std::unique_ptr<PoolFileEntry>> pool_;
```

Comment:
please put this in a class with a static method with static variable as we have in there parts of the codebase.
this will ensure correct order of initialisation on use, not related to linking order. 
the mutex is then a member of that class that you can lock and unlock as you insert and remove entries to the map via proper defined methods